### PR TITLE
Masked token from request_uri and referrer in nginx ingress logs

### PR DESCRIPTION
**Pull request by rujhan-arora-astronomer**
## Description

Mask the `token` query param in referrer and request url string, in our nginx logs

## Related Issues

[#5608
](https://github.com/astronomer/issues/issues/5608)
## Testing

Local kind cluster

## Merging

0.32.1

### CODE REVIEW

File: charts/nginx/templates/nginx-configmap.yaml [hunk 1]
```diff
@@ -25,7 +25,81 @@ data:
...
+        local queryString = string.sub(url, queryStringStart + 1)
+        local queryParams = {}
+
+        -- Split the query string into individual parameters
+        for param in string.gmatch(queryString, "([^&]+)") do
+          local paramName, paramValue = string.match(param, "(.+)=(.+)")
+
+          -- Check if the parameter should be masked
+          if paramsToMask[paramName] then
+              paramValue = "********" -- Replace the value with asterisks
```
Comment (pgvishnuram) [inline charts/nginx/templates/nginx-configmap.yaml:55]: we can keep same message as REDACTED how we show for astroUI logs

File: charts/nginx/templates/nginx-configmap.yaml [hunk 2]
```diff
@@ -25,7 +25,81 @@ data:
...
```
Comment (pgvishnuram) [inline charts/nginx/templates/nginx-configmap.yaml:92]: we can keep same message as REDACTED how we show for astroUI logs

**Comment by rujhan-arora-astronomer** [comment]
@pgvishnuram Done, replaced with REDACTED

**Review by rishkarajgi — APPROVED** [review]

File: charts/nginx/templates/nginx-configmap.yaml [hunk 3]
```diff
@@ -25,7 +25,81 @@ data:
...
+        local queryString = string.sub(url, queryStringStart + 1)
+        local queryParams = {}
+
+        -- Split the query string into individual parameters
+        for param in string.gmatch(queryString, "([^&]+)") do
+          local paramName, paramValue = string.match(param, "(.+)=(.+)")
+
+          -- Check if the parameter should be masked
+          if paramsToMask[paramName] then
+              paramValue = "REDACTED" -- Replace the value with asterisks
```
Comment (rishkarajgi) [inline charts/nginx/templates/nginx-configmap.yaml:92]: comment needs to be updated `Replace the value with asterisks`

File: charts/nginx/templates/nginx-configmap.yaml [hunk 4]
```diff
@@ -52,7 +52,7 @@ data:
 
           -- Check if the parameter should be masked
           if paramsToMask[paramName] then
-              paramValue = "********" -- Replace the value with asterisks
+              paramValue = "REDACTED" -- Replace the value with asterisks
```
Comment (danielhoherd) [inline charts/nginx/templates/nginx-configmap.yaml:55]:
```suggestion
              paramValue = "REDACTED" -- Replace sensitive data with 'REDACTED'
```

**Review by danielhoherd — APPROVED** [review]
Seems good to me. This isn't the kind of thing we can test in this repo so we'll need to test it in E2E tests.

**Pull request merged**